packages/context/src/context.ts

```diff
@@ -121,28 +121,38 @@ export class Context {
 
   /**
    * Find bindings using the key pattern
-   * @param pattern Key regexp or pattern with optional `*` wildcards
+   * @param pattern A regexp or pattern with optional `*` wildcards. If it
+   * matches the binding key, the binding is included.
    */
-  find(pattern?: string | RegExp): Binding[] {
+  find(pattern?: string | RegExp): Binding[];
+
+  /**
+   * Find bindings using a filter function
+   * @param filter A function to test on the binding. It returns `true` to
+   * include the binding or `false` to exclude the binding.
+   */
+  find(filter: (binding: Binding) => boolean): Binding[];
+
+  find(pattern?: string | RegExp | ((binding: Binding) => boolean)): Binding[] {
     let bindings: Binding[] = [];
-    let glob: RegExp | undefined = undefined;
-    if (typeof pattern === 'string') {
-      // TODO(@superkhau): swap with production grade glob to regex lib
```

Comment:
Please preserve this comment. Your current solution `new RegExp('^' + pattern.split('*').join('.*') + '$')` will not correctly handle edge cases, for example:

 - the string pattern `foo$bar` will not match anything, because`$` is interpreted as the end of the input.
 - the string pattern `foo.bar`will match `fooxbar` too, because `.` is interpreted as "any character " instead of "dot character"

Reply:
I fixed the code to use strict `wildcard` respecting `.` or `:` as separators.
- `*`: matching zero or more chars excluding `.` or `:`
- `?`: matching exactly one char excluding `.` or `:`